### DISCUSSION TITLE
Can only build DDEMO effect

### DISCUSSION THREAD

**Discussion by ScottCLT** [PlummersSoftwareLLC/NightDriverStrip#54] [Q&A]
First off, I am not a programmer - I'm a tinkerer. So bear with me if this is basic. But I'm unable to build any effects except for DDEMO. It works and the scrolling rainbow effect looks beautiful! But if I try to switch to DATOMLIGHT, DSPECTRUM, etc. I get the following build errors, even though I'm using a Heltec module that I purchased from the amazon link in the description.

I searched for the header files "M5StickCPlus.h" and "M5StickC.h" and found some header files online from M5stack and included those, but I get the same build errors, below. Any pointers on getting this to build?

Building in debug mode
Compiling .pio\build\heltec\src\audio.cpp.o
Compiling .pio\build\heltec\src\colordata.cpp.o
Compiling .pio\build\heltec\src\drawing.cpp.o
Compiling .pio\build\heltec\src\effects.cpp.o
Compiling .pio\build\heltec\src\main.cpp.o
Compiling .pio\build\heltec\src\network.cpp.o
Compiling .pio\build\heltec\src\screen.cpp.o
Compiling .pio\build\heltec\src\uzlib\src\tinflate.c.o
Compiling .pio\build\heltec\src\uzlib\src\tinfzlib.c.o
Generating partitions .pio\build\heltec\partitions.bin
Compiling .pio\build\heltec\lib29e\IRremoteESP8266\IRac.cpp.o
src/screen.cpp: In function 'void UpdateScreen()':
src/screen.cpp:151:17: error: 'M5' was not declared in this scope
                 M5.Lcd.fillScreen(BLACK16);
                 ^
In file included from src/screen.cpp:36:0:
include/freefonts.h:145:15: error: 'FreeSans9pt7b' was not declared in this scope
 #define FF17 &FreeSans9pt7b
               ^
src/screen.cpp:153:36: note: in expansion of macro 'FF17'
                 M5.Lcd.setFreeFont(FF17);
                                    ^
src/screen.cpp:158:37: error: 'C_BASELINE' was not declared in this scope
                 M5.Lcd.setTextDatum(C_BASELINE);
                                     ^
In file included from src/screen.cpp:36:0:
include/freefonts.h:146:15: error: 'FreeSans12pt7b' was not declared in this scope
 #define FF18 &FreeSans12pt7b
               ^
src/screen.cpp:166:36: note: in expansion of macro 'FF18'
                 M5.Lcd.setFreeFont(FF18);
                                    ^
include/freefonts.h:125:14: error: 'FreeMono9pt7b' was not declared in this scope
 #define FF1 &FreeMono9pt7b
              ^
src/screen.cpp:180:40: note: in expansion of macro 'FF1'
                     M5.Lcd.setFreeFont(FF1);
                                        ^
include/freefonts.h:135:14: error: 'FreeMonoOblique9pt7b' was not declared in this scope
 #define FF9 &FreeMonoOblique9pt7b
              ^
src/screen.cpp:184:36: note: in expansion of macro 'FF9'
                 M5.Lcd.setFreeFont(FF9);
                                    ^
src/screen.cpp:208:17: error: 'M5' was not declared in this scope
                 M5.Lcd.drawString(szBuffer, 0, 13); // write something to the internal memory
                 ^
src/screen.cpp:229:13: error: 'M5' was not declared in this scope
             M5.Lcd.drawString(szBuffer, 0, 37); // write something to the internal memory
             ^
*** [.pio\build\heltec\src\screen.cpp.o] Error 1

**Comment by imCharlieB** [comment]
all the effects are there if you want to change effects for the demo open the effects.cpp file its in src folder and go down to the if DEMO section should be around line 269ish and you can add some different effects i added these when i was testing the demo out `    // Animate a simple rainbow palette by using the palette effect on the built-in rainbow palette
    new PaletteEffect(rainbowPalette, 256/16, .2, 0),
    new ClassicFireEffect(),
    new BouncingBallEffect(),
    new SimpleRainbowTestEffect(8, 1),  // Rainbow palette simple test of walking pixels
    new TwinkleEffect(NUM_LEDS/2, 0, 50),
    new RainbowFillEffect(),`
